### DISCUSSION TITLE
Any way to set global mermaid chunk options?

### DISCUSSION THREAD

**Discussion by cjdbarlow** [quarto-dev/quarto-cli#7279] [Q&A] [answered]
### Description

I would like to define chunk options for styling all mermaid diagrams created in the document, rather than having to manually do it each time. For example, I would like all graphs to use the `stepBefore` curve style. I can achieve this manually with:

````
```{mermaid}
%%{ init: { 'flowchart': { 'curve': 'stepBefore' } } }%%

graph_goes_here
```
````

But this distracts from the focus of the source document, which should be (imo) content rather than styling. I note similar things have been [raised before](https://github.com/quarto-dev/quarto-cli/discussions/6209) so I am not sure if this is doable? Is there a way to prepend text to different code chunks?

**Comment by jimjam-slam** [comment]
I'm not sure if you can directly inject Mermaid defaults into document YAML, but you _can_ either [set a theme](https://quarto.org/docs/authoring/diagrams.html#mermaid-theming) or [override the theme defaults](https://quarto.org/docs/authoring/diagrams.html#customizing-mermaid) from the YAML using SASS variables that Quarto provides. That might not be enough for non-CSS configuration options like using `stepBefore`, though!

**Reply by cjdbarlow** [reply]
Thanks @jimjam-slam - you're right that they only allow for CSS configuration though. As far as I can work out, they also only work when you are serving the native JS and the rendering occurs client-side, and I'd like to display the graphs both when javascript is disabled and a non-HTML output format is used.

**Comment by cscheid** [comment]
This is something we could add to the baseline mermaid feature. We have our plates very full at the moment, but I'l create an enhancement ticket for it.

**Reply by cscheid** [reply, marked as answer]
See https://github.com/quarto-dev/quarto-cli/issues/7295
